utils/src/main/java/org/apache/cloudstack/utils/security/DigestHelper.java

```diff
@@ -131,4 +140,14 @@ protected static boolean isAlgorithmPresent(String checksum) {
     public static String getHashValueFromChecksumValue(String checksum) {
         return isAlgorithmPresent(checksum) ? new ChecksumValue(checksum).getChecksum() : checksum;
     }
+
+    public static String calculateChecksum(File file) {
+        try (InputStream is = Files.newInputStream(Paths.get(file.getPath()))) {
+            return DigestUtils.md5Hex(is);
+        } catch (IOException e) {
+            String errMsg = "Failed to calculate template checksum";
+            LOGGER.error(errMsg, e);
```

Comment:
rather no stacktraces on error level. my take on this is if you want it log the message twice and add the trace to a debug level message. Also the exception is being re-thrown so we need to think about whether the stacktrace is needed at this point.